src/main/webapp/app/exercises/shared/exercise-headers/header-exercise-page-with-details.component.ts

```diff
@@ -72,4 +91,63 @@ export class HeaderExercisePageWithDetailsComponent implements OnChanges {
             this.icon = getIcon(exerciseType) as IconProp;
         }
     }
+
+    /**
+     * Determines what element of the header should be highlighted. The highlighted deadline/time is the one being due next
+     */
+    private setIsNextDueDateExamMode() {
+        this.isNextDueDate = [false, false];
```

Comment:
Although this solution seems to work, it is quite hard to comprehend for me. There is no association between the slot in the array and the date it is referring to. In order to understand this logic, one would need to find the usage of the array positions in the HTML file first. Maybe consider adding some description to make it clearer for other developers. :)

Reply:
Will add some jsdoc